canvas/models/models.py

```diff
@@ -175,9 +176,9 @@ def username(self):
 
     @property
     def name(self):
-        if self.user.has_name:
-            return self.user.get_full_name()
-        return "Anonymous Student"
+        if self.user.has_nickname:
```

Comment:
but it can't be null right?

Reply:
Technically, but this is for the very worst case

Reply:
But if we have the integrity constraint on the DB, then it's not the worst case, it's an impossible case, no?